lecture_notes/2017-02-14.md

```diff
@@ -0,0 +1,362 @@
+# Title: Linear Types for Low-level Languages
+## Speaker: Dan Patterson
+
+Goal today: show how linear types can help make programs run more efficiently.
+
+This is active research! A bunch of "recent" commercial and academic languages
+use linear types:
+- [Rust](https://www.rust-lang.org/), see the recent
+  [keynote](http://popl17.sigplan.org/event/popl-2017-papers-rust-from-popl-to-practice#modal-phid_a19ef757dfad82a6b45b08606914533d)
+  ([pdf](./resources/t-popl-2017.pdf)) at POPL 2017.
+- [Cogent](https://arxiv.org/abs/1601.05520), see the recent
+  [paper}(http://icfp16.sigplan.org/event/icfp-2016-papers-refinement-through-restraint-bringing-down-the-cost-of-verification)
+  ([pdf](http://people.mpi-inf.mpg.de/~crizkall/Publications/cogent.pdf)) at ICFP 2016.
+- [ATS](http://www.ats-lang.org/), developed by Hongwei Xi's group at Boston University.
+- [Cyclone](http://cyclone.thelanguage.org/)
+  ([wiki](https://en.wikipedia.org/wiki/Cyclone_(programming_language))),
+  no longer supported, but was previously maintained by Greg Morrisett's group at Cornell.
+- [Vault](https://swehb.nasa.gov/display/7150/Vault), previously maintained by Manuel Fähndrich.
+  See the [PLDI 2001](./resources/df-pldi-2001.pdf)
+  and [PLDI 2002](./resources/fd-pldi-2002.pdf) papers.
+
+> M: (about Rust's keynote) you know what Landin said: as soon as the
+>    theoreticians get ahold of an idea, it's all downhill.
+
+> J: why did Cyclone fail?
+> D: good question
+> J: maybe **the** question (perhaps, something that cannot be overcome)
+> D: well Rust credits Cyclone as a major influence
+> M: academic projects don't last. That's it.
+
+### Fundamentals
+
+#### Sequent Calculus
+
+Girard introduced linear logic in a 1987 paper.
+Interestingly, there's a remark in the paper that lienar logic could be used
+to speed up computations.
```

Comment:
Yes - "Linear logic will therefore help us improve the efficiency of programs." This was in the context of computer science, were resource usage actually mattered, prompted this form of logic (he wrote, "computer science prompted the subject through semantics").